examples/behavior_robot_example.cpp

```diff
@@ -0,0 +1,122 @@
+ *        constructs the control loop, which contains a behavior manager, for a
+ *        `SimpleRobot` and provides several behaviors which can be switched
+ *        between via LCM broadcast.
+ * @date 7/27/22
+ * 
+ * @copyright Copyright 2022 The Trustees of the University of Pennsylvania. All
+ *            rights reserved.
+ * 
+ */
+
+#include <vector>  // std::vector
+#include "ManyMotorLog.hpp"  // ManyMotorLog
+#include "ModeInput.hpp"  // ModeInput
+#include "kodlab_mjbots_sdk/mjbots_behavior_loop.h"  // kodlab::mjbots::MjbotsBehaviorLoop
+#include "examples/simple_robot.h"  // SimpleRobot
+#include "examples/simple_spin_joints_behavior.h"  // SimpleSpinJointsBehavior
+#include "examples/simple_control_io_behavior.h" // SimpleControlIOBehavior
+
+/**
+ * @brief Behavior loop for a simple robot
+ * @note Unlike the `SimpleRobotControlLoop` provided in `robot_example.cpp`,
+ *       this control loop does not override the `Update` method. This is
+ *       handled internally by the `MjbotsBehaviorLoop`.
+ */
+class SimpleRobotBehaviorLoop : public kodlab::mjbots::MjbotsBehaviorLoop<
+    ManyMotorLog, ModeInput, SimpleRobot>
+{
+  /**
+   * @brief Use `MjbotsBehaviorLoop` constructor
+   */
+  using MjbotsBehaviorLoop::MjbotsBehaviorLoop;
+
+  /**
+   * @brief Logs joint state information
+   * @details This method is called every control loop update, and logs general
+   *          joint data to a `ManyMotorLog` via the LCM log data topic.
+   */
+  void PrepareLog() override
+  {
+    for (int servo = 0; servo < num_joints_; servo++)
+    {
+      log_data_->positions[servo] = robot_->GetJointPositions()[servo];
+      log_data_->velocities[servo] = robot_->GetJointVelocities()[servo];
+      log_data_->modes[servo] =
+          static_cast<int>(mjbots_interface_->GetJointModes()[servo]);
+      log_data_->torques[servo] = robot_->GetJointTorqueCmd()[servo];
+    }
+    for (int servo = num_joints_; servo < 13; servo++)
+    {
+      log_data_->positions[servo] = 0;
+      log_data_->velocities[servo] = 0;
+      log_data_->modes[servo] = 0;
+      log_data_->torques[servo] = 0;
+    }
+  }
+
+  /**
+   * @brief Callback for new mode input LCM input message
+   * @details This function passes input mode selection integers received via
+   *          LCM to the behavior loop's internal behavior manager. The behavior
+   *          manager then handles the actual behavior switching.
+   */
+  void ProcessInput(const ModeInput &input_data) override
+  {
+    // Set input behavior on internal behavior manager
+    behavior_mgr.SetBehavior(input_data.mode);
+  }
+};
+
+/**
+ * @brief Main method
+ * @return 0 if program exited without issue
+ */
+int main(int argc, char **argv)
+{
+  // Setup joints with a std::vector
+  std::vector<kodlab::mjbots::JointMoteus> joints;
+  joints.emplace_back(22, 1, 1, 0, 1.0, 5.0);
+//  joints.emplace_back(101, 4, -1, 0, 5.0 / 3.0, 0);
```

Comment:
remove this probably

Reply:
Fixed in 8cecc842b6606aa38becbff419a2a1fc14f11dbb.